flow/project.py

```diff
@@ -1660,12 +1660,19 @@ def _execute_operation(self, operation, timeout=None):
             or operation.name not in self._operation_functions
             # The specified executable is not the same as the interpreter instance:
             or operation.directives.get('executable', sys.executable) != sys.executable
+            # the operation requires mpi or omp_num_thread
+            or operation.directives.get('nranks', 1) > 1
+            or operation.directives.get('omp_num_threads', 1) > 1
         ):
             # ... need to fork:
+            if operation.directives['nranks']:
+                prefix = self._environment.get_prefix(operation)
+            else:
+                prefix = ''
             logger.debug(
                 "Forking to execute operation '{}' with "
-                "cmd '{}'.".format(operation, operation.cmd))
-            subprocess.run(operation.cmd, shell=True, timeout=timeout, check=True)
+                "cmd '{}'.".format(operation, prefix + operation.cmd))
+            subprocess.run(prefix + operation.cmd, shell=True, timeout=timeout, check=True)
```

Comment:
`prefix + operation.cmd` is dangerous. We then have to be sure that `prefix` always has an ending space. We should add a space between just to be sure.